spock-specs/src/test/groovy/org/spockframework/smoke/Interceptors.groovy

```diff
@@ -154,54 +154,105 @@ class FooSpec extends Specification {
     @Override
     void visitSpecAnnotation(LifecycleTest annotation, SpecInfo specInfo) {
       specInfo.addSharedInitializerInterceptor {
+        it.with {
```

Comment:
What triggered this addition? And can we reduce the duplication to either a one-liner or move it inside of proceed.

Reply:
I'm not sure what you mean with "this addition". Do you mean the `it.with`? It is in preparation of the other assertions coming into the same block.

Regarding putting it into `proceed`, `spec` is special because it should be set in all cases. But I thought it would be better to keep these assertions grouped there with the assertions from the upstack PRs.